### ISSUE TITLE
What should behaviour be when a block is very long?

### ISSUE THREAD

**Issue by jwzimmer-zz** [LLK/scratch-blocks#940]
### Expected Behavior

However large my block is, should I be able to somehow see all of it when I drag it, either by scrolling or some other approach?

Should there be a max length/ max size? I would guess not, because it's easy to imagine that if you're using e.g. operators or custom blocks you could get some very long blocks.

### Actual Behavior

When the block is long enough that it's wider than my computer screen, when I'm dragging it, I can't scroll or drag it any further to the left in order to see the rest of the block.

<img width="1145" alt="screen shot 2017-06-12 at 2 10 05 pm" src="https://user-images.githubusercontent.com/1320456/27048483-07b61390-4f7a-11e7-9ab1-f19a85f4e7e8.png">

### Steps to Reproduce

- Make a very long block
- Click & drag it to make it come to the front layer

### Operating System and Browser

_Mac OS El Capitan, Chrome 59_

On https://llk.github.io/scratch-gui/.

**Comment by rachel-fenichel** [comment]
Are you testing this on llk.github.io/scratch-gui?

**Comment by rachel-fenichel** [comment]
If so, is the screenshot above mid-block-drag?

**Comment by ericrosenbaum** [comment]
Another example:

![big-move](https://user-images.githubusercontent.com/567844/27049102-4540ac00-4f7c-11e7-8645-7f09d696cde8.gif)

**Comment by rachel-fenichel** [comment]
@ericrosenbaum that doesn't seem like the same bug.  That's #430.

My question is about "When the block is long enough that it's wider than my computer screen, I can't scroll or drag or anything in order to see the rest of the block.",

because I'm able to scroll the workspace to see all parts of the block.  I'm trying to understand whether @jwzimmer is unable to scroll the workspace when there's a large block, or if the issue is just that at some point you can't see everything that is typed at the same time.

**Comment by jwzimmer-zz** [comment]
@rachel-fenichel, yes, this is mid-drag & on https://llk.github.io/scratch-gui/ - I was trying to move the block in such a way that I could see the ends of the block (not necessarily all at once, probably by scrolling). I was thinking of how in Scratch 2.0, if you drag a block to the edge of the scripting area, the scripting area will automatically scroll for you - I was thinking of something like that happening here to show me the rest of the block. However, that might not be the behaviour we want to implement - the current behaviour might be the best way to deal with a super long block; I'm not sure.

**Comment by rachel-fenichel** [comment]
Okay, so the question is whether dragging a block to the edge of the workspace bumps the workspace to make it scroll in the direction of your drag?

If that's the question, cool, I'll leave it up to Scratch-team to make the decision.  I was concerned that you meant that scrolling and dragging was broken if you had a single block with a very long input.

**Comment by jwzimmer-zz** [comment]
Yes, @rachel-fenichel, that's the question - scrolling and dragging are not broken for the long block, as far as I noticed. 👍
